xc7/utils/db_overlay/db_overlay.py

```diff
@@ -0,0 +1,98 @@
+"""
+This file contains a "proxy" class which implements an "overlay" for the
+prjxray database.
+"""
+
+import os
+import json
+
+from prjxray import db
+from prjxray import tile
+
+# =============================================================================
+
+
+class DatabaseWithOverlay(db.Database):
+    """
+    Prjxray database overlay
+    """
+
+    def __init__(self, db_root, overlay_root=None):
+        """
+        Constructor.
+
+        :param db_root: Prjxray database root
+        :param db_overlay: Prjxray database overlay root
+        """
+
+        # Initialize base class
+        db.Database.__init__(self, db_root)
+
+        # Get overlay root
+        if overlay_root is None:
+            overlay_root = os.path.dirname(__file__)
+        self.overlay_root = overlay_root
+
+        # Load additional tile types
+        for f in os.listdir(self.overlay_root):
+            if f.endswith('.json') and f.startswith('tile_type_'):
+                tile_type = f[len('tile_type_'):-len('.json')].lower()
+
+                print("Overlay tile type: '%s'" % tile_type)
+
+                segbits = os.path.join(
```

Comment:
There shouldn't be any segbits, pips or mask files in the overlay, so why is this here?

Reply:
I just copied reading of all relevant tile data. It is not necessary so I'll remove it.

Reply:
Ok, removed that.